### PR TITLE
Don't Log First Exception when Connecting to Proxy

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Fixes https://github.com/Azure/azure-sdk-for-java/issues/30148

Updates Netty proxy handling to not throw an exception on the first attempt to connect to the proxy. Credentials aren't applied on the first request so it is common for this to fail.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/CallMetadata.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * Contains metadata about a request call.
+ */
+public final class CallMetadata {
+    private volatile Boolean firstCall = true;
+    private static final AtomicReferenceFieldUpdater<CallMetadata, Boolean> FIRST_CALL_UPDATER
+        = AtomicReferenceFieldUpdater.newUpdater(CallMetadata.class, Boolean.class, "firstCall");
+
+    private volatile Boolean firstCallWithProxy = false;
+    private static final AtomicReferenceFieldUpdater<CallMetadata, Boolean> FIRST_CALL_WITH_PROXY_UPDATER
```
Comment (lmolkova) [inline sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/CallMetadata.java:17]: I learned something today!

**Review by lmolkova — DISMISSED** [review]

**Comment by alzimmermsft** [comment]
Running live test for Storage is showing OutOfMemoryErrors which need to be investigated further before merging

**Comment by alzimmermsft** [comment]
> Hey! We still need this fix. It would be nice if it wasn't lost.

Wasn't lost! Just been bogged down with a few other tasks. I'm getting back around to this now 😄

**Review by lmolkova — APPROVED** [review]

**Comment by nandorsoma** [comment]
> > Hey! We still need this fix. It would be nice if it wasn't lost.
> 
> Wasn't lost! Just been bogged down with a few other tasks. I'm getting back around to this now 😄

No problem, I just commented to remove the flag. Thanks for pulling it off!

**Pull request merged**